cpp/src/io/parquet/compact_protocol_reader.cpp

```diff
@@ -439,24 +450,25 @@ class parquet_field_optional : public parquet_field {
  */
 void CompactProtocolReader::skip_struct_field(int t, int depth)
 {
-  switch (t) {
-    case ST_FLD_TRUE:
-    case ST_FLD_FALSE: break;
-    case ST_FLD_I16:
-    case ST_FLD_I32:
-    case ST_FLD_I64: get_u64(); break;
-    case ST_FLD_BYTE: skip_bytes(1); break;
-    case ST_FLD_DOUBLE: skip_bytes(8); break;
-    case ST_FLD_BINARY: skip_bytes(get_u32()); break;
-    case ST_FLD_LIST: [[fallthrough]];
-    case ST_FLD_SET: {
+  auto const t_enum = static_cast<FieldType>(t);
+  switch (t_enum) {
+    case FieldType::BOOLEAN_TRUE:
+    case FieldType::BOOLEAN_FALSE: break;
+    case FieldType::I16:
+    case FieldType::I32:
+    case FieldType::I64: get_u64(); break;
+    case FieldType::I8: skip_bytes(1); break;
+    case FieldType::DOUBLE: skip_bytes(8); break;
+    case FieldType::BINARY: skip_bytes(get_u32()); break;
+    case FieldType::LIST: [[fallthrough]];
```

Comment:
Is there a recommendation on when to use `[[fallthrough]]`?  I'm not sure it's used consistently here.
Would it be wrong to simply remove the `[[fallthrough]]` here?  We don't seem to have used it above.

Reply:
The wording I've always seen is something to the effect that multiple cases for a block do not require `[[falthrough]]`. It's really intended for the case:
```c++
  case A:
    { // do some stuff }
    [[fallthrough]]; // let compiler know the lack of a `break` is intentional
  case B:
    { // do more stuff }
    break;
```

Reply:
lol, first time reading this guideline. Makes sense.
I tend to use `[[fallthrough]]` for multiple cases as well. I thought compiler might warn for any missing breaks (eventually, or with some flags).

Reply:
Fixed, thanks for pointing this out. My fallthrough tyranny has ended.